cosmo_tester/test_suites/manager_of_managers/regional_cluster.py

```diff
@@ -0,0 +1,984 @@
+        blueprint_file = self._cloned_to / self.blueprint_file
+
+        self.logger.info('Uploading blueprint: %s [id=%s]',
+                         namespace_blueprint_file,
+                         namespace_blueprint_id)
+        self.logger.info('Uploading blueprint: %s [id=%s]',
+                         blueprint_file,
+                         self.blueprint_id)
+
+        if use_cfy:
+            self.cfy.profile.set([
+                '-t', self.tenant,
+            ])
+            self.cfy.blueprint.upload([
+                '-b', namespace_blueprint_id,
+                namespace_blueprint_file
+            ])
+            self.cfy.blueprint.upload([
+                '-b', self.blueprint_id,
+                blueprint_file
+            ])
+        else:
+            if self.first_deployment:
+                with util.set_client_tenant(self.manager, self.tenant):
+                    self.manager.client.blueprints.upload(
+                        namespace_blueprint_file, namespace_blueprint_id)
+                    self.manager.client.blueprints.upload(
+                        blueprint_file, self.blueprint_id)
+
+    def install(self):
+        super(AbstractRegionalCluster, self).install()
+        self._populate_status_output()
+
+    def _populate_status_output(self):
+        # This workflow populates the deployment outputs with status info
+        self.cfy.executions.start('get_status', '-d', self.deployment_id)
+        self.cfy.deployments.outputs(self.deployment_id)
+
+    def verify_installation(self):
+        super(AbstractRegionalCluster, self).verify_installation()
+        assert self.master_ip
+
+    def deploy_and_validate(self):
+        if self._deployed:
+            self.logger.info('Regional cluster was already deployed')
+            return
+        self.logger.info(
+            'Deploying Regional cluster on deployment: {0}'.format(
+                self.deployment_id
+            )
+        )
+        self._deployed = True
+        self.upload_and_verify_install()
+        self.validate()
+
+    def backup(self):
+        self.logger.info(
+            'Running backup workflow on Regional '
+            'cluster on dep: {0}...'.format(
+                self.deployment_id
+            )
+        )
+
+        backup_params = {
+            'snapshot_id': self.deployment_id,
+            'backup_params': []
+        }
+        self.cfy.executions.start(
+            'backup', '-d', self.deployment_id,
+            '-p', json.dumps(backup_params)
+        )
+        self.logger.info('Backup completed successfully')
+
+    def scale(self):
+        self.logger.info('Scaling deployment...')
+        self._cleanup_required = True
+        try:
+            self.cfy.executions.start.scale(
+                ['-d', self.deployment_id,
+                 '-p', 'scalable_entity_name=workers_group',
+                 '-p', 'delta=1',
+                 '-t', self.tenant, '-vv'])
```

Comment:
Are we sure to keep -vv?

Reply:
I don't understand why `-vv` is not the default everywhere. These are tests. Need debug logging.

Reply:
because we save all the logs from the manager, I don't know if that feature also supports this